sys/flash_map/src/flash_map.c

```diff
@@ -131,6 +131,73 @@ flash_area_getnext_sector(int id, int *sec_id, struct flash_area *ret)
     return rc;
 }
 
+int
+flash_area_to_subareas(int id, int *cnt, struct flash_area *ret)
+{
+    int rc;
+    int i, j;
+    const struct hal_flash *hf;
+    const struct flash_area *fa;
+    int max_cnt;
+    int next;
+    int first_idx;
+    int last_idx;
+    uint32_t start;
+    uint32_t size;
+    uint32_t min_size;
+
+    rc = flash_area_open(id, &fa);
+    if (rc != 0) {
+        return rc;
+    }
+    first_idx = last_idx = -1;
+    max_cnt = *cnt;
+    *cnt = 0;
+
+    hf = hal_bsp_flash_dev(fa->fa_device_id);
+    for (i = 0; i < hf->hf_sector_cnt; i++) {
+        hf->hf_itf->hff_sector_info(hf, i, &start, &size);
+        if (start >= fa->fa_off && start < fa->fa_off + fa->fa_size) {
+            if (first_idx == -1) {
+                first_idx = i;
+            }
+            last_idx = i;
+            *cnt = *cnt + 1;
```

Comment:
looks like *cnt is assigned in line 196, so line 165 is not really needed

Reply:
On close inspection, realized that it is used as an intermediate variable, and used in line 168.

Reply:
Maybe adding local variable for this would make it simpler and probably would reduce code size as well.